lib/vdc/BOVCollection.cpp

```diff
@@ -473,18 +549,17 @@ template<class T> int BOVCollection::ReadRegion(std::string varname, size_t ts,
         return -1;
     }
 
-    size_t numValues = _gridSize[0] * _gridSize[1] * _gridSize[2];
-
     int  n = 1;
     bool systemLittleEndian = *(char *)&n == 1 ? true : false;
-    bool dataLittleEndian = _dataEndian == "LITTLE" ? true : false;
+    bool dataLittleEndian = _dataEndian == _littleEndianString ? true : false;
     bool needSwap = systemLittleEndian != dataLittleEndian ? true : false;
```

Comment:
Why don't you write it in a simpler form (and more intuitive):
```c++
bool dataLittleEndian = _dataEndian == _littleEndianString;
bool needSwap = systemLittleEndian != dataLittleEndian;
```

Reply:
Endian stuff has been removed.